sycl/include/CL/sycl/detail/buffer_impl.hpp

```diff
@@ -66,35 +65,30 @@ template <typename AllocatorT> class buffer_impl {
   buffer_impl(const void *hostData, const size_t sizeInBytes,
               const property_list &propList,
               AllocatorT allocator = AllocatorT())
-      : SizeInBytes(sizeInBytes), Props(propList), MAllocator(allocator) {
-    if (Props.has_property<property::buffer::use_host_ptr>()) {
+      : SizeInBytes(sizeInBytes), Props(propList),
+        BufPtr(const_cast<void *>(hostData)), MAllocator(allocator) {
```

Comment:
Is the cast to void necessary?

Reply:
It doesn't looks good, but yes.

Reply:
Ah I see. I missed the `const` in the constructor.